### ISSUE TITLE
No results are being shown in HUB when running a model training session on a local system

### ISSUE THREAD

**Issue by jens-siebert** [ultralytics/ultralytics#21990]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train, Other

### Bug

I'm running a YOLO11 model training session on a local system.

In the past, when logging in to the Ultralytics HUB from my systems first, I was able to view the intermediate and final results of the training in the HUB.

This doesn't seem to work anymore as I can see an entry for the newly trained model but the results view doesn't show any data nor does it update the number of remaining epochs. It shows "Last updated a few seconds ago", though.

### Environment

OS => Ubuntu 24.04.3 LTS
Browser => Firefox 142.0

Ultralytics 8.3.195 🚀 Python-3.12.3 torch-2.8.0+cu128 CUDA:0 (NVIDIA GeForce RTX 2080 Super, 7754MiB)
Setup complete ✅ (20 CPUs, 62.7 GB RAM, 454.6/912.8 GB disk)

OS                     Linux-6.14.0-29-generic-x86_64-with-glibc2.39
Environment            Linux
Python                 3.12.3
Install                git
Path                   /home/jens/Documents/Projekte/doks/rack_boundary_detector/venv/lib/python3.12/site-packages/ultralytics
RAM                    62.74 GB
Disk                   454.6/912.8 GB
CPU                    Intel Core(TM) i9-10900K 3.70GHz
CPU count              20
GPU                    NVIDIA GeForce RTX 2080 Super, 7754MiB
GPU count              1
CUDA                   12.8

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.6>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 11.3.0>=7.1.2
pyyaml                 ✅ 6.0.2>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.16.1>=1.4.1
torch                  ✅ 2.8.0>=1.8.0
torch                  ✅ 2.8.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.23.0>=0.9.0
psutil                 ✅ 7.0.0
polars                 ✅ 1.33.0
ultralytics-thop       ✅ 2.0.17>=2.0.0

### Minimal Reproducible Example

```python
from ultralytics import YOLO, hub

hub.login("<api_key>")
model = YOLO("yolo11s.pt")
model.train(data="<dataset_yaml", epochs=250, imgsz=640, patience=100)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jens-siebert, thanks for reporting this and for using Ultralytics HUB with YOLO11 🚀 This is an automated response to help you get moving quickly—an Ultralytics engineer will also assist you here soon.

Please review the [Docs](https://docs.ultralytics.com/) for common usage guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
- Since you’re seeing missing/empty results in Ultralytics HUB for a local training run, please also include:
  - Exact command(s) or script used and full console logs from start to finish (including any HUB sync messages) 📋
  - Confirmation that you’re logged in to Ultralytics HUB in the same environment (e.g., `hub.login("<api_key>")` as shown)
  - Whether the issue occurs consistently across multiple runs or datasets
  - Any network/firewall/VPN constraints that could block outbound requests
  - The HUB project name/run entry you see created (text description is fine)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks for the detailed environment info and the snippet you shared. Once you can attach the requested logs and any additional reproduction details, we’ll be able to investigate the HUB sync behavior more quickly 🙏

**Comment by Y-T-G** [comment]
@jens-siebert Does it work with Ultralytics v8.3.193?

**Comment by ghost** [comment]
I can confirm that with your fixes, the model training data in the HUB does update again. Thanks for fixing 👍

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in the PR [Fix metrics logging to HUB](https://github.com/ultralytics/ultralytics/pull/21993) — big thanks to @Y-T-G for the implementation and to @glenn-jocher for the contributions.

What changed and why this may resolve the issue:
- The Ultralytics HUB session is now passed directly through training args during Trainer initialization, ensuring the session is available immediately.
- Trainer extracts the HUB session early from overrides and removes the previous late assignment path.
- This reduces side effects and improves reliability of metrics logging, syncing, and callbacks with Ultralytics HUB.

How to test the fix:
- Install the latest code from main:
  ```
  pip install -U git+https://github.com/ultralytics/ultralytics.git@main
  ```
- Or wait for the next official PyPI release and then upgrade:
  ```
  pip install -U ultralytics
  ```
- After updating, run a short training while connected to Ultralytics HUB and confirm that metrics and artifacts appear and update as expected in your HUB project.

Feedback requested:
- Please share whether the merged changes resolve the issue in your environment.
- If problems persist, include a brief reproduction (command used), logs, and environment details to help investigate further.

Thanks for reporting this and helping improve reliability 🙏
